tools/gfx/persistent-shader-cache.cpp

```diff
@@ -42,39 +48,69 @@ void PersistentShaderCache::loadCacheFromFile()
     ComPtr<ISlangBlob> indexBlob;
     if (SLANG_FAILED(desc.shaderCacheFileSystem->loadFile(desc.cacheFilename, indexBlob.writeRef())))
     {
-        // Cache index not found, so we'll create and save a new one.
-        if (mutableShaderCacheFileSystem)
+        // Cache index not found and we can't save a new one due to the file system being immutable.
+        if (!mutableShaderCacheFileSystem)
         {
-            mutableShaderCacheFileSystem->saveFile(desc.cacheFilename, nullptr, 0);
             return;
         }
-        // Cache index not found and we can't save a new one due to the file system being immutable.
-        return;
+        // Cache index not found, so we'll create and save a new one.
+        mutableShaderCacheFileSystem->saveFile(desc.cacheFilename, nullptr, 0);
     }
-
-    auto indexString = UnownedStringSlice((char*)indexBlob->getBufferPointer());
-
-    List<UnownedStringSlice> lines;
-    StringUtil::calcLines(indexString, lines);
-    for (auto line : lines)
+    else
     {
-        List<UnownedStringSlice> digests;
-        StringUtil::split(line, ' ', digests); // This will return our two hashes as two elements in digests, unless we've reached the end.
-        if (digests.getCount() != 2)
-            continue;
-        auto dependencyDigest = DigestUtil::fromString(digests[0]);
-        auto contentsDigest = DigestUtil::fromString(digests[1]);
-
-        ShaderCacheEntry entry = { dependencyDigest, contentsDigest };
-        auto entryNode = entries.AddLast(entry);
-        keyToEntry.Add(dependencyDigest, entryNode);
-
-        // If there are more entries present in the cache file than the entry count limit allows,
-        // ignore the rest. Since we output the lines in the cache file in order of most to least
-        // recently used, the cache's behavior will still be correct.
-        if (desc.entryCountLimit > 0 && entries.Count() == desc.entryCountLimit)
-            break;
+        auto indexString = UnownedStringSlice((char*)indexBlob->getBufferPointer());
```

Comment:
I suggest we do binary file read write to skip the whole parsing/toString conversion altogether.